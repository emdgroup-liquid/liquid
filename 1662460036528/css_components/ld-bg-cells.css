.ld-bg-cells,:host{--ld-bg-cells-bg-col:var(--ld-thm-secondary);--ld-bg-cells-layer-col:var(--ld-thm-primary);--ld-bg-cells-layer-translation-x:-80%;--ld-bg-cells-layer-translation-y:-10%;--ld-bg-cells-layer-size:260%;--ld-bg-cells-layer-rotation:0deg;--ld-bg-cells-secondary-layer-col:#0000;--ld-bg-cells-secondary-layer-translation-x:0%;--ld-bg-cells-secondary-layer-translation-y:0%;--ld-bg-cells-secondary-layer-size:150%;--ld-bg-cells-secondary-layer-rotation:0deg;--ld-bg-cells-base-size-factor:7;--ld-bg-cells-layer-size-normalized:calc(clamp(100%, var(--ld-bg-cells-layer-size), 800%)*var(--ld-bg-cells-base-size-factor));--ld-bg-cells-secondary-layer-size-normalized:calc(clamp(100%, var(--ld-bg-cells-secondary-layer-size), 800%)*var(--ld-bg-cells-base-size-factor));background:var(--ld-bg-cells-bg-col);height:100%;overflow:hidden;position:relative;width:100%}.ld-bg-cells--three-layers,:host(.ld-bg-cells--three-layers){--ld-bg-cells-layer-col:var(--ld-col-vy);--ld-bg-cells-secondary-layer-col:var(--ld-thm-primary);--ld-bg-cells-layer-size:580%;--ld-bg-cells-layer-translation-x:-20%;--ld-bg-cells-layer-translation-y:-100%;--ld-bg-cells-secondary-layer-size:390%;--ld-bg-cells-secondary-layer-translation-x:-50%;--ld-bg-cells-secondary-layer-rotation:-30deg}.ld-bg-cells--bioreliance,:host(.ld-bg-cells--bioreliance){--ld-bg-cells-base-size-factor:8.4;--ld-bg-cells-bg-col:var(--ld-col-vg);--ld-bg-cells-layer-col:var(--ld-col-rp);--ld-bg-cells-secondary-layer-col:#0000;--ld-bg-cells-layer-translation-x:-89%;--ld-bg-cells-layer-translation-y:85%;--ld-bg-cells-layer-size:297%}.ld-bg-cells--f,:host(.ld-bg-cells--f){--ld-bg-cells-base-size-factor:8.7;--ld-bg-cells-layer-translation-x:-79%;--ld-bg-cells-layer-translation-y:43%;--ld-bg-cells-layer-size:240%}.ld-bg-cells--millipore,:host(.ld-bg-cells--millipore){--ld-bg-cells-base-size-factor:7.8;--ld-bg-cells-bg-col:var(--ld-col-rb);--ld-bg-cells-layer-col:var(--ld-col-vy);--ld-bg-cells-secondary-layer-col:#0000;--ld-bg-cells-layer-translation-x:-80%;--ld-bg-cells-layer-translation-y:78%;--ld-bg-cells-layer-size:230%}.ld-bg-cells--milliq,:host(.ld-bg-cells--milliq){--ld-bg-cells-base-size-factor:4.2;--ld-bg-cells-bg-col:var(--ld-col-vc);--ld-bg-cells-layer-col:var(--ld-col-rp);--ld-bg-cells-secondary-layer-col:#0000;--ld-bg-cells-layer-translation-x:-70%;--ld-bg-cells-layer-translation-y:180%;--ld-bg-cells-layer-size:420%}.ld-bg-cells--o,:host(.ld-bg-cells--o){--ld-bg-cells-base-size-factor:6;--ld-bg-cells-layer-translation-x:-75%;--ld-bg-cells-layer-translation-y:-38%;--ld-bg-cells-layer-size:190%}.ld-bg-cells--supelco,:host(.ld-bg-cells--supelco){--ld-bg-cells-base-size-factor:12;--ld-bg-cells-bg-col:var(--ld-col-rg);--ld-bg-cells-layer-col:var(--ld-col-vy);--ld-bg-cells-secondary-layer-col:#0000;--ld-bg-cells-layer-translation-x:-59%;--ld-bg-cells-layer-translation-y:76%;--ld-bg-cells-layer-size:190%}.ld-bg-cells--safc,:host(.ld-bg-cells--safc){--ld-bg-cells-base-size-factor:6;--ld-bg-cells-bg-col:var(--ld-col-vm);--ld-bg-cells-layer-col:var(--ld-col-vy);--ld-bg-cells-secondary-layer-col:#0000;--ld-bg-cells-layer-translation-x:-122%;--ld-bg-cells-layer-translation-y:8%;--ld-bg-cells-layer-size:362%}.ld-bg-cells--sigma-aldrich,:host(.ld-bg-cells--sigma-aldrich){--ld-bg-cells-base-size-factor:9.4;--ld-bg-cells-bg-col:var(--ld-col-rr);--ld-bg-cells-layer-col:var(--ld-col-vy);--ld-bg-cells-secondary-layer-col:#0000;--ld-bg-cells-layer-translation-x:-124%;--ld-bg-cells-layer-translation-y:-4%;--ld-bg-cells-layer-size:460%}.ld-bg-cells--t,:host(.ld-bg-cells--t){--ld-bg-cells-base-size-factor:8.4;--ld-bg-cells-layer-translation-x:-108%;--ld-bg-cells-layer-translation-y:129%;--ld-bg-cells-layer-size:312%}.ld-bg-cells--tile,:host(.ld-bg-cells--tile){--ld-bg-cells-base-size-factor:5.6;--ld-bg-cells-layer-translation-x:-93%;--ld-bg-cells-layer-translation-y:-2%;--ld-bg-cells-layer-size:340%}.ld-bg-cells__layer{background:var(--ld-bg-cells-layer-col);transform:translate(var(--ld-bg-cells-layer-translation-x),var(--ld-bg-cells-layer-translation-y)) scale(var(--ld-bg-cells-layer-size-normalized)) rotate(var(--ld-bg-cells-layer-rotation))}.ld-bg-cells__layer,.ld-bg-cells__secondary-layer{-webkit-mask-image:var(--ld-bg-cells-image);mask-image:var(--ld-bg-cells-image);-webkit-mask-position:center;mask-position:center;-webkit-mask-repeat:no-repeat;mask-repeat:no-repeat}.ld-bg-cells__secondary-layer{background:var(--ld-bg-cells-secondary-layer-col);-webkit-mask-composite:xor;mask-composite:exclude;transform:translate(var(--ld-bg-cells-secondary-layer-translation-x),var(--ld-bg-cells-secondary-layer-translation-y)) scale(var(--ld-bg-cells-secondary-layer-size-normalized)) rotate(var(--ld-bg-cells-secondary-layer-rotation))}.ld-bg-cells__layer,.ld-bg-cells__secondary-layer{inset:0;position:absolute}