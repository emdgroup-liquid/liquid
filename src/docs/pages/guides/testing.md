---
eleventyNavigation:
  key: Testing
  parent: Guides
  order: 10
layout: layout.njk
title: Testing
permalink: guides/testing/
---

# Testing

If you want to __unit test__ code in your app, which depends on the functionality of Liquid Oxygen Web Components, without mocking their internals, then you must make sure all relevant components have _hydrated_ in your test setup, before you execute your test code. You can do so by running the `defineCustomElements()` method imported from `'@emdgroup-liquid/liquid/dist/loader'` before each test and a `await`ing a helper function, which makes sure all components involved in the test have hydrated.

The following example shows how to execute unit tests with hydrated Web Components with [vitest](https://vitest.dev/), React, and the [React Testing Library](https://testing-library.com/docs/react-testing-library/intro/).

First, [configure vitest to use a setup file](https://vitest.dev/config/#setupfiles):

```js
// vite.config.ts
import { defineConfig } from 'vitest/config'
export default defineConfig({
  // ...
  test: {
    // ...
    setupFiles: './src/setupTests.ts',
  },
})
```

The `setupTests.ts` file looks like this:

```js
// src/setupTests.ts
import { defineCustomElements } from '@emdgroup-liquid/liquid/dist/loader'
import '@testing-library/jest-dom'
import matchers from '@testing-library/jest-dom/matchers'
import { cleanup } from '@testing-library/react'
import { vi } from 'vitest'

// Extend Vitest's expect method with methods from react-testing-library.
expect.extend(matchers)

beforeAll(() => {
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  global.__LD_ASSET_PATH__ = '/' // removes CDN warning
  defineCustomElements()

  global.fetch = vi
    .fn()
    .mockReturnValue(Promise.resolve({ text: () => Promise.resolve('') }))
})

// Run a cleanup after each test case.
afterEach(() => {
  cleanup()
})
```

In your test file, you can now wait for the Web Components to hydrate before executing your actual test code.

```tsx
// src/App.test.tsx
import App from './App'
import { render, screen, waitFor, fireEvent } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'

async function waitForLiquidToHydrate() {
  return waitFor(() => {
    Array.from(document.querySelectorAll('*'))
      .filter((e) => e.tagName.startsWith('LD-'))
      .forEach((component) => {
        expect(component).toHaveClass('hydrated')
      })
  })
}

test('validates input', async () => {
  render(<App />)

  await waitForLiquidToHydrate()

  const ldInputName = screen.getByPlaceholderText('e.g. Jason Parse')
  expect(ldInputName).toBeInTheDocument()

  const ldButtonSubmit = screen.getByText('Submit')
  expect(ldButtonSubmit).toBeInTheDocument()
  userEvent.click(ldButtonSubmit)

  const form = await screen.findByTestId('form')
  expect(form).toBeInTheDocument()

  fireEvent.submit(form)

  await screen.findByText('Your full name is required.')
})
```

Check out our [React sandbox app](https://stackblitz.com/github/emdgroup-liquid/liquid-sandbox-react-tailwind?file=src%2FApp.test.tsx) for a full working example.

<docs-page-nav prev-href="guides/design-tokens/" next-title="Sandbox applications" next-href="guides/sandbox-applications/"></docs-page-nav>

