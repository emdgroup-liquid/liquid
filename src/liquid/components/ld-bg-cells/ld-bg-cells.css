.ld-bg-cells {
  background: var(--ld-bg-cells-background-color);
  height: 100%;
  position: relative;
  width: 100%;

  :where(&--bioreliance) {
    --ld-bg-cells-position: top 5% right -20%;
    --ld-bg-cells-size: 230%;
  }

  :where(&--f) {
    --ld-bg-cells-position: top 20% right -50%;
    --ld-bg-cells-size: 180%;
  }

  :where(&--hexagon) {
    --ld-bg-cells-position: top 56% right -30%;
    --ld-bg-cells-size: 200%;
  }

  :where(&--millipore) {
    --ld-bg-cells-position: top -10% right -30%;
    --ld-bg-cells-size: 200%;
  }

  :where(&--qa-x2f-qc) {
    --ld-bg-cells-position: top -30% right -34%;
    --ld-bg-cells-size: 170%;
  }

  :where(&--safc) {
    --ld-bg-cells-position: top 47% right -15%;
    --ld-bg-cells-size: 290%;
  }

  :where(&--sigma-aldrich) {
    --ld-bg-cells-position: top 51% right -19%;
    --ld-bg-cells-size: 280%;
  }

  :where(&--t) {
    --ld-bg-cells-position: top -10% right -20%;
    --ld-bg-cells-size: 250%;
  }

  :where(&--tile) {
    --ld-bg-cells-position: top 50% right -12%;
    --ld-bg-cells-size: 250%;
  }
}

.ld-bg-cells__pattern {
  background: var(--ld-bg-cells-foreground-color);
  mask-image: var(--ld-bg-cells-image);
  mask-position: var(--ld-bg-cells-position);
  mask-repeat: no-repeat;
  mask-size: var(--ld-bg-cells-size);

  /* Results in pixelated pattern */
  /* &--repeat {
    mask-repeat: repeat;
  } */
}

.ld-bg-cells__pattern,
.ld-bg-cells__content {
  inset: 0;
  position: absolute;
}

/* .ld-theme-ocean -> default */
.ld-bg-cells,
:where(.ld-theme-ocean) .ld-bg-cells,
:where([class*='ld-theme'] .ld-theme-ocean) .ld-bg-cells {
  --ld-bg-cells-background-color: var(--ld-thm-ocean-bg-primary);
  --ld-bg-cells-foreground-color: var(--ld-thm-ocean-accent);
}

:where(.ld-theme-tea),
:where([class*='ld-theme'] .ld-theme-tea) {
  .ld-bg-cells {
    --ld-bg-cells-background-color: var(--ld-thm-tea-bg-primary);
    --ld-bg-cells-foreground-color: var(--ld-thm-tea-accent);
  }
}

:where(.ld-theme-bubblegum),
:where([class*='ld-theme'] .ld-theme-bubblegum) {
  .ld-bg-cells {
    --ld-bg-cells-background-color: var(--ld-thm-bubblegum-bg-primary);
    --ld-bg-cells-foreground-color: var(--ld-thm-bubblegum-accent);
  }
}

:where(.ld-theme-shake),
:where([class*='ld-theme'] .ld-theme-shake) {
  .ld-bg-cells {
    --ld-bg-cells-background-color: var(--ld-thm-shake-bg-primary);
    --ld-bg-cells-foreground-color: var(--ld-thm-shake-accent);
  }
}

:where(.ld-theme-solvent),
:where([class*='ld-theme'] .ld-theme-solvent) {
  .ld-bg-cells {
    --ld-bg-cells-background-color: var(--ld-thm-solvent-bg-primary);
    --ld-bg-cells-foreground-color: var(--ld-thm-solvent-accent);
  }
}
