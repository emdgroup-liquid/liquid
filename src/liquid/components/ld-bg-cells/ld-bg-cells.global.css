:where(.ld-theme-ocean),
:where([class*='ld-theme'] .ld-theme-ocean) {
  .ld-bg-cells {
    --ld-bg-cells-background-color: var(--ld-thm-ocean-bg-primary);
    --ld-bg-cells-foreground-color: var(--ld-thm-ocean-accent);
  }
}

:where(.ld-theme-tea),
:where([class*='ld-theme'] .ld-theme-tea) {
  .ld-bg-cells {
    --ld-bg-cells-background-color: var(--ld-thm-tea-bg-primary);
    --ld-bg-cells-foreground-color: var(--ld-thm-tea-accent);
  }
}

:where(.ld-theme-bubblegum),
:where([class*='ld-theme'] .ld-theme-bubblegum) {
  .ld-bg-cells {
    --ld-bg-cells-background-color: var(--ld-thm-bubblegum-bg-primary);
    --ld-bg-cells-foreground-color: var(--ld-thm-bubblegum-accent);
  }
}

:where(.ld-theme-shake),
:where([class*='ld-theme'] .ld-theme-shake) {
  .ld-bg-cells {
    --ld-bg-cells-background-color: var(--ld-thm-shake-bg-primary);
    --ld-bg-cells-foreground-color: var(--ld-thm-shake-accent);
  }
}

:where(.ld-theme-solvent),
:where([class*='ld-theme'] .ld-theme-solvent) {
  .ld-bg-cells {
    --ld-bg-cells-background-color: var(--ld-thm-solvent-bg-primary);
    --ld-bg-cells-foreground-color: var(--ld-thm-solvent-accent);
  }
}
