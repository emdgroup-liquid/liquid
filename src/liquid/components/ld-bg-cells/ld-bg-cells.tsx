import { Build, Component, Element, h, Host, Prop, Watch } from '@stencil/core'
import { getClassNames } from '../../utils/getClassNames'
import { fetchPattern } from '../../utils/fetchAsset'

export type CellType =
  | 'bioreliance'
  | 'f' // Functional
  | 'functional'
  | 'hexagon' // Synthetic
  | 'mdo'
  | 'millipore'
  | 'milliq'
  | 'o' // Organic
  | 'organic'
  | 'plastic'
  | 'qa-x2f-qc' // Supelco
  | 'safc'
  | 'sigma-aldrich'
  | 'supelco'
  | 'synthetic'
  | 't' // Technical
  | 'technical'
  | 'tile' // Plastic

/**
 * @virtualProp ref - reference to component
 * @virtualProp {string | number} key - for tracking the node's identity when working with lists
 * @part layer - the primary cell layer
 * @part secondary-layer - the secondary cell layer
 */
@Component({
  assetsDirs: ['assets'],
  tag: 'ld-bg-cells',
  styleUrl: 'ld-bg-cells.css',
  shadow: true,
})
export class LdBgCells {
  @Element() el: HTMLElement

  /** Cells pattern */
  @Prop() type: CellType = 'hexagon'

  /** Use 3 color layers */
  @Prop() threeLayers? = false

  /** Animate the pattern */
  @Prop() animated? = false

  @Watch('type')
  private async loadPatternPathData(): Promise<void> {
    if ((!Build.isBrowser && !Build.isTesting) || !this.type) {
      return
    }

    const patternString = await fetchPattern(this.type)
    this.el.shadowRoot.querySelectorAll('svg').forEach((layer) => {
      const div = document.createElement('div')
      div.innerHTML = patternString
      Array.from(div.children[0]?.children || []).forEach((child) => {
        if (child.tagName !== 'script') {
          layer.appendChild(child)
        }
      })
    })
  }

  componentWillLoad() {
    this.loadPatternPathData()
  }

  render() {
    // Handle aliases (for backward compatibility).
    let cellType = this.type

    if (cellType === 'qa-x2f-qc') cellType = 'supelco'
    if (cellType === 'functional') cellType = 'f'
    if (cellType === 'technical') cellType = 't'
    if (cellType === 'plastic') cellType = 'tile'
    if (cellType === 'synthetic') cellType = 'hexagon'
    if (cellType === 'organic') cellType = 'o'

    return (
      <Host
        class={getClassNames([
          'ld-bg-cells',
          `ld-bg-cells--${cellType}`,
          this.threeLayers && 'ld-bg-cells--three-layers',
        ])}
      >
        <svg
          class={getClassNames([
            'ld-bg-cells__secondary-layer',
            this.animated && 'ld-bg-cells__secondary-layer--animated',
          ])}
          viewBox="0 0 8000 8000"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
          part="secondary-layer"
        ></svg>
        <svg
          class={getClassNames([
            'ld-bg-cells__layer',
            this.animated && 'ld-bg-cells__layer--animated',
          ])}
          viewBox="0 0 8000 8000"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
          part="layer"
        ></svg>
      </Host>
    )
  }
}
