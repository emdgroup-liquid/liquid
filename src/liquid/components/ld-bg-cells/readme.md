---
eleventyNavigation:
  key: Background Cells
  parent: Components
layout: layout.njk
title: Background Cells
permalink: components/ld-bg-cells/
---

<!-- <link rel="stylesheet" href="/css_components/ld-bg-cells.css"> -->

# ld-bg-cells

A background pattern with the Merck cells as additional visual element.

---

## Default

{% example %}
<ld-bg-cells class="box1-16/9"></ld-bg-cells>

<style>
  .box1-16\/9 {
    padding-top: 56.25%;
  }
</style>
{% endexample %}

<!-- CSS component -->

<!-- <div class="ld-bg-cells ld-bg-cells--safc box1-16/9">
  <div class="ld-bg-cells__pattern"></div>
</div>

<style>
  .box1-16\/9 {
    padding-top: 56.25%;
  }
</style> -->

## Cell types

{% example %}
<ld-bg-cells type="bioreliance" class="box2-16/9">
  <ld-paragraph>type: bioreliance</ld-paragraph>
</ld-bg-cells>
<ld-bg-cells type="f" class="box2-16/9">
  <ld-paragraph>type: f</ld-paragraph>
</ld-bg-cells>
<ld-bg-cells type="hexagon" class="box2-16/9">
  <ld-paragraph>type: hexagon</ld-paragraph>
</ld-bg-cells>
<ld-bg-cells type="millipore" class="box2-16/9">
  <ld-paragraph>type: millipore</ld-paragraph>
</ld-bg-cells>
<ld-bg-cells type="qa-x2f-qc" class="box2-16/9">
  <ld-paragraph>type: qa-x2f-qc</ld-paragraph>
</ld-bg-cells>
<ld-bg-cells type="safc" class="box2-16/9">
  <ld-paragraph>type: safc</ld-paragraph>
</ld-bg-cells>
<ld-bg-cells type="sigma-aldrich" class="box2-16/9">
  <ld-paragraph>type: sigma-aldrich</ld-paragraph>
</ld-bg-cells>
<ld-bg-cells type="t" class="box2-16/9">
  <ld-paragraph>type: t</ld-paragraph>
</ld-bg-cells>
<ld-bg-cells type="tile" class="box2-16/9">
  <ld-paragraph>type: tile</ld-paragraph>
</ld-bg-cells>

<style>
  .box2-16\/9 {
    height: 12.3046875rem;
    width: 21.875rem;
  }

  .box2-16\/9 .ld-bg-cells__content {
    align-items: center;
    color: var(--ld-col-wht);
    display: flex;
    justify-content: center;
  }
</style>
{% endexample %}

<!-- CSS component -->

<!-- <div class="ld-bg-cells ld-bg-cells--bioreliance box2-16/9">
  <div class="ld-bg-cells__pattern"></div>
  <div class="ld-bg-cells__content">
    <p class="ld-paragraph">type: bioreliance</p>
  </div>
</div>
<div class="ld-bg-cells ld-bg-cells--f box2-16/9">
  <div class="ld-bg-cells__pattern"></div>
  <div class="ld-bg-cells__content">
    <p class="ld-paragraph">type: f</p>
  </div>
</div>
<div class="ld-bg-cells ld-bg-cells--hexagon box2-16/9">
  <div class="ld-bg-cells__pattern"></div>
  <div class="ld-bg-cells__content">
    <p class="ld-paragraph">type: hexagon</p>
  </div>
</div>
<div class="ld-bg-cells ld-bg-cells--millipore box2-16/9">
  <div class="ld-bg-cells__pattern"></div>
  <div class="ld-bg-cells__content">
    <p class="ld-paragraph">type: millipore</p>
  </div>
</div>
<div class="ld-bg-cells ld-bg-cells--qa-x2f-qc box2-16/9">
  <div class="ld-bg-cells__pattern"></div>
  <div class="ld-bg-cells__content">
    <p class="ld-paragraph">type: qa-x2f-qc</p>
  </div>
</div>
<div class="ld-bg-cells ld-bg-cells--safc box2-16/9">
  <div class="ld-bg-cells__pattern"></div>
  <div class="ld-bg-cells__content">
    <p class="ld-paragraph">type: safc</p>
  </div>
</div>
<div class="ld-bg-cells ld-bg-cells--sigma-aldrich box2-16/9">
  <div class="ld-bg-cells__pattern"></div>
  <div class="ld-bg-cells__content">
    <p class="ld-paragraph">type: sigma-aldrich</p>
  </div>
</div>
<div class="ld-bg-cells ld-bg-cells--t box2-16/9">
  <div class="ld-bg-cells__pattern"></div>
  <div class="ld-bg-cells__content">
    <p class="ld-paragraph">type: t</p>
  </div>
</div>
<div class="ld-bg-cells ld-bg-cells--tile box2-16/9">
  <div class="ld-bg-cells__pattern"></div>
  <div class="ld-bg-cells__content">
    <p class="ld-paragraph">type: tile</p>
  </div>
</div>

<style>
  .box2-16\/9 {
    height: 12.3046875rem;
    width: 21.875rem;
  }

  .box2-16\/9 .ld-bg-cells__content {
    align-items: center;
    color: var(--ld-col-wht);
    display: flex;
    justify-content: center;
  }
</style> -->

## Custom position, size and colors

{% example %}
<ld-bg-cells class="box3-16/9"></ld-bg-cells>

<style>
  .box3-16\/9 {
    --ld-bg-cells-bg-col: var(--ld-col-vm-200);
    --ld-bg-cells-pattern-col: var(--ld-col-vm-400);
    --ld-bg-cells-position: bottom left;
    --ld-bg-cells-size: 150%;

    padding-top: 56.25%;
  }
</style>
{% endexample %}

<!-- CSS component -->

<!-- <div class="ld-bg-cells ld-bg-cells--safc box3-16/9">
  <div class="ld-bg-cells__pattern"></div>
</div>

<style>
  .box3-16\/9 {
    --ld-bg-cells-bg-col: var(--ld-col-vm-200);
    --ld-bg-cells-pattern-col: var(--ld-col-vm-400);
    --ld-bg-cells-position: bottom left;
    --ld-bg-cells-size: 150%;

    padding-top: 56.25%;
  }
</style> -->

## CSS Variables

| Variable                         | Description                                                   | Default                              |
| -------------------------------- | ------------------------------------------------------------- | ------------------------------------ |
| `--ld-bg-cells-bg-col` | Background color                                              | Primary color of the current theme   |
| `--ld-bg-cells-pattern-col` | Pattern color (cells)                                      | Secondary color of the current theme |
| `--ld-bg-cells-position`         | Position of the cells                                         | Individual per cell type             |
| `--ld-bg-cells-size`             | Size of the cells                                             | Individual per cell type             |


<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description                                              | Type                                                                                                            | Default     |
| -------- | --------- | -------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------- | ----------- |
| `key`    | `key`     | for tracking the node's identity when working with lists | `string \| number`                                                                                              | `undefined` |
| `ref`    | `ref`     | reference to component                                   | `any`                                                                                                           | `undefined` |
| `type`   | `type`    | Cells pattern                                            | `"bioreliance" \| "f" \| "hexagon" \| "millipore" \| "qa-x2f-qc" \| "safc" \| "sigma-aldrich" \| "t" \| "tile"` | `'safc'`    |


## Shadow Parts

| Part        | Description                          |
| ----------- | ------------------------------------ |
| `"content"` | Element wrapping the slot            |
| `"pattern"` | Element containing the cells pattern |


----------------------------------------------

 
