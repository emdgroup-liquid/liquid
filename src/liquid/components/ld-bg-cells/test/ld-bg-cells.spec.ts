import { newSpecPage } from '@stencil/core/testing'
import { LdBgCells } from '../ld-bg-cells'

describe('ld-bg-cells', () => {
  it('renders default', async () => {
    const page = await newSpecPage({
      components: [LdBgCells],
      html: `<ld-bg-cells></ld-bg-cells>`,
    })
    expect(page.root).toMatchSnapshot()
  })

  it('renders with explicit type', async () => {
    const page = await newSpecPage({
      components: [LdBgCells],
      html: `<ld-bg-cells type="millipore"></ld-bg-cells>`,
    })
    expect(page.root).toMatchSnapshot()
  })

  it('renders with alias type', async () => {
    const page = await newSpecPage({
      components: [LdBgCells],
      html: `<ld-bg-cells type="qa-x2f-qc"></ld-bg-cells>`,
    })
    expect(page.root).toMatchSnapshot()
  })

  it('renders with three layers', async () => {
    const page = await newSpecPage({
      components: [LdBgCells],
      html: `<ld-bg-cells three-layers></ld-bg-cells>`,
    })
    expect(page.root).toMatchSnapshot()
  })

  it('renders with animation', async () => {
    const page = await newSpecPage({
      components: [LdBgCells],
      html: `<ld-bg-cells animated></ld-bg-cells>`,
    })
    expect(page.root).toMatchSnapshot()
  })
})
