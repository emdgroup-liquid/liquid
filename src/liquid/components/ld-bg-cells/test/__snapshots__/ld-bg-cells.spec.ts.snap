// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ld-bg-cells renders default 1`] = `
<ld-bg-cells class="ld-bg-cells ld-bg-cells--safc" style="--ld-bg-cells-image: url(/assets/safc-cell.svg);">
  <!---->
  <div class="ld-bg-cells__pattern"></div>
  <div class="ld-bg-cells__content"></div>
</ld-bg-cells>
`;

exports[`ld-bg-cells renders with content 1`] = `
<ld-bg-cells class="ld-bg-cells ld-bg-cells--safc" style="--ld-bg-cells-image: url(/assets/safc-cell.svg);">
  <!---->
  <div class="ld-bg-cells__pattern"></div>
  <div class="ld-bg-cells__content">
    <p>
      I am content
    </p>
  </div>
</ld-bg-cells>
`;

exports[`ld-bg-cells renders with explicit type 1`] = `
<ld-bg-cells class="ld-bg-cells ld-bg-cells--millipore" type="millipore" style="--ld-bg-cells-image: url(/assets/millipore-cell.svg);">
  <!---->
  <div class="ld-bg-cells__pattern"></div>
  <div class="ld-bg-cells__content"></div>
</ld-bg-cells>
`;

exports[`ld-bg-cells renders with repeat 1`] = `
<ld-bg-cells class="ld-bg-cells ld-bg-cells--safc" repeat="" style="--ld-bg-cells-image: url(/assets/safc-cell.svg);">
  <!---->
  <div class="ld-bg-cells__pattern"></div>
  <div class="ld-bg-cells__content"></div>
</ld-bg-cells>
`;
